server/controllers/medical/patients/merge.js

```diff
@@ -0,0 +1,102 @@
+  `;
+  const replaceDebtorInCash = `
+    UPDATE cash SET debtor_uuid = ? WHERE debtor_uuid IN (?);
+  `;
+  const replaceDebtorInDebtorGroupHistory = `
+    UPDATE debtor_group_history SET debtor_uuid = ? WHERE debtor_uuid IN (?);
+  `;
+  const replaceDebtorInInvoice = `
+    UPDATE invoice SET debtor_uuid = ? WHERE debtor_uuid IN (?);
+  `;
+  const replaceDebtorInPatient = `
+    UPDATE patient SET debtor_uuid = ? WHERE debtor_uuid IN (?);
+  `;
+  const replacePatientInEmployee = `
+    UPDATE employee SET patient_uuid = ? WHERE patient_uuid IN (?);
+  `;
+  const replacePatientInPatientAssignment = `
+    UPDATE patient_assignment SET patient_uuid = ? WHERE patient_uuid IN (?);
+  `;
+  const replacePatientInPatientDocument = `
+    UPDATE patient_document SET patient_uuid = ? WHERE patient_uuid IN (?);
+  `;
+  const replacePatientInPatientHospitalization = `
+    UPDATE patient_hospitalization SET patient_uuid = ? WHERE patient_uuid IN (?);
+  `;
+  const replacePatientInPatientVisit = `
+    UPDATE patient_visit SET patient_uuid = ? WHERE patient_uuid IN (?);
+  `;
+
+  const removeOtherDebtors = `
+    DELETE FROM debtor WHERE uuid IN (?);
+  `;
+  const removeOtherPatients = `
+    DELETE FROM patient WHERE uuid IN (?);
+  `;
+
+  db.one(getSelectedDebtorUuid, [selectedPatientUuid])
+    .then(row => {
+      glb.selectedDebtorUuid = row.debtor_uuid;
+      return db.exec(getOtherDebtorUuids, otherPatientUuids);
+    })
+    .then(rows => {
+      glb.otherDebtorUuids = rows.map(row => row.debtor_uuid);
+
+      const transaction = db.transaction();
+
+      transaction.addQuery(replaceDebtorInJournal, [glb.selectedDebtorUuid, [glb.otherDebtorUuids]]);
+      transaction.addQuery(replaceDebtorInLedger, [glb.selectedDebtorUuid, [glb.otherDebtorUuids]]);
+
+      transaction.addQuery(replaceDebtorInCash, [glb.selectedDebtorUuid, [glb.otherDebtorUuids]]);
+      transaction.addQuery(replaceDebtorInDebtorGroupHistory, [glb.selectedDebtorUuid, [glb.otherDebtorUuids]]);
+      transaction.addQuery(replaceDebtorInInvoice, [glb.selectedDebtorUuid, [glb.otherDebtorUuids]]);
+      transaction.addQuery(replaceDebtorInPatient, [glb.selectedDebtorUuid, [glb.otherDebtorUuids]]);
+
+      transaction.addQuery(replacePatientInEmployee, [selectedPatientUuid, [otherPatientUuids]]);
+      transaction.addQuery(replacePatientInPatientAssignment, [selectedPatientUuid, [otherPatientUuids]]);
+      transaction.addQuery(replacePatientInPatientDocument, [selectedPatientUuid, [otherPatientUuids]]);
+      transaction.addQuery(replacePatientInPatientHospitalization, [selectedPatientUuid, [otherPatientUuids]]);
+      transaction.addQuery(replacePatientInPatientVisit, [selectedPatientUuid, [otherPatientUuids]]);
+
+      transaction.addQuery(removeOtherPatients, [otherPatientUuids]);
+      transaction.addQuery(removeOtherDebtors, [glb.otherDebtorUuids]);
+      return transaction.execute();
```

Comment:
Since we merge just two patients not more, i think i can replace all these codes by a MySQL procedure.

The reason of these staff are just in the case we will have to merge more than one patients into a one, it is not obvious to send an array of uuid to a MySQL procedure.